superset/views/core.py

```diff
@@ -1200,17 +1186,30 @@ def annotation_json(self, layer_id):
             form_data=form_data,
             force=False,
         )
-        try:
-            payload = viz_obj.get_payload()
-        except Exception as e:
-            logging.exception(e)
-            return json_error_response(utils.error_msg_from_exception(e))
-        status = 200
-        if payload.get('status') == QueryStatus.FAILED:
-            status = 400
-        return json_success(viz_obj.json_dumps(payload), status=status)
+        payload_json_and_error = viz_obj.get_payload_json_and_error()
+        return self.payload_response(payload_json_and_error)
 
     @log_this
+    @api
+    @handle_superset_exception
+    @has_access_api
+    @expose('/api/v1/query/', methods=['POST'])
```

Comment:
Would be nice to do this in a different file under `views/api.py` maybe

Reply:
Yep, can do.

Reply:
Yeah, I did a quick look around as well for how much needs to be moved to have a proper `api` folder under `app`. It is a non-trivial amount and probably not a priority to tackle right now. 

I'll stick with  `{ROOT}/views/api.py` for now.